src/controller/CHIPDeviceController.cpp

```diff
@@ -559,6 +551,11 @@ void DeviceCommissioner::ReleaseCommissioneeDevice(CommissioneeDeviceProxy * dev
     {
         mDeviceBeingCommissioned = nullptr;
     }
+
+    // Release the commissionee device after we have nulled out our pointers,
+    // because that can call back in to us with error notifications as the
+    // session is released.
+    mCommissioneeDevicePool.ReleaseObject(device);
```

Comment:
@bzbarsky-apple Looks like there are two calls now to ReleaseObject from the device pool. I am getting an error when I tried to use commissioning through the tv-casting-app. Is it expected to have two calls to ReleaseObject?

Reply:
This got fixed in #22403.